build/link-wrapper.sh

```diff
@@ -2,6 +2,11 @@
 
 set -e
 
+if [ -z ${CC+x} ]
+then
+    CC=gcc
+fi
+
```

Comment:
This works in the other docker image, because we do:

```
ARG CC=/usr/bin/gcc
ENV CC=$CC
```

which doesn't make me the happiest.

The proper way to handle this would be to go to Makefile, and after the platform stanzas that optionally set things about each platform, do:

```
ifeq ($(origin CC), default)
CC := $(CC)
endif
```

to actually push the defaulted $CC into the environment for the scripts that we call, and if it's overridden in the environment, then we leave it alone.  (`?=` doesn't work for anything with a default value)

Reply:
I agree that this is ugly, but my line of thinking was more: if we plan to deprecate this build system soon anyways, is it really worth to solve all these things properly?

Reply:
Alright, I'll change it myself.